### ISSUE TITLE
ESP-01S Heap Configuration

### ISSUE THREAD

**Issue by PixlRainbow** [JiriBilek/ESP_ATMod#10]
I'm using a generic ESP-01S module plugged in to an FTDI with Arduino IDE, and I've mostly left the settings on defaults except for setting "Debug" to "Serial".
I just cloned this repo today and flashed it to my module, and while it mostly seems to work fine, it appears to have trouble with verifying certificates. It fails with an error when I call `AT+CIPSSLCERT`, and based on the source it appears to be because **it couldn't allocate the 4096 bytes needed on the heap to fit the PEM buffer**.
Next, I tried adding the pem CA file via littleFS upload, and `AT+CIPSSLCERT?` query seemed to indicate that one certificate had loaded successfully. But when I attempt to query `AT+CIPSSLCERT?1`, it fails again with a non-descriptive error without printing out the cert details.
Finally, I tried turning on certificate chain verification and connecting to a test openssl s_server. It said "connect fail" and then "CLOSED". It was able to connect to the same server when chain verification was disabled.
Testing directly with openssl s_client on a PC with chain verification enabled and configured to use the cert as a CA file indicated that the server could be verified successfully.

**I'm not sure if I'm supposed to be configuring Arduino IDE any differently to change the heap size/allocation algorithm, or if my particular module simply has less RAM than the ones you've tested.**

Testing `AT+SYSRAM?` starting with default CIPSSLAUTH value on an SSL (aka TLSv1.2) connection:
Before connection: 43712
Active connection: 16032
After closing connection: 43480
After set `AT+CIPSSLAUTH=2`: 43480
After set `AT+CIPSSLAUTH=0`: 43536

EDIT: also asking, which version of the esp8266 core are you testing with?

**Comment by JiriBilek** [comment]
Hi, thanks for the feedback.

I found a bug in the `AT+CIPSSLCERT` command and I have just comitted a fix. It fixes both the issue with loading a new certificate and with querying a particular one. Please test it.

And regarding the SSL connection: did you configure SNTP time before opening a connection? The command is `AT+CIPSNTPCFG`. The certificate verification requires the time to be set.

**Comment by PixlRainbow** [comment]
I just pulled d6ad63f and flashed it to my ESP-01S. I can now load new certificates through AT terminal and query specific certificates. Interestingly, instead of displaying the root CA name as shown in the readme example, it simply says `cert OK`.

I have configured the SNTP time to my time zone and left it on default time servers. I queried `AT+CIPSNTPTIME?` to confirm that the time matched the time on my PC. It is still unable to perform chain verification successfully. Enabling SSL debug shows the messages:
```
BSSL:_connectSSL: start connection
BSSL:_wait_for_handshake: failed
BSSL:Couldn't connect. Error = 'Expected server name was not found in the chain.'
```
Stackexchange suggests that this was because the common name either matched with the issuer name (self-signed certificate) or that the common name (CN) did not explicitly list the fully qualified domain name.
I have created my own root certificate authority and then signed a separate server certificate using it, making sure that the CA cert had a CN set, that the CN of the CA was different from the server cert, and that the server cert had common name matching the LAN FQDN (`starlight-desktop.local`), with the LAN IP address as subject alternative name for backup. I made sure both the root CA cert and the server cert were loaded onto the firmware, and that `AT+CIPSTART` was called using the FQDN, not the IP address, though I also attempted using IP address with the same result.
